scipy/stats/_continuous_distns.py

```diff
@@ -4195,10 +4195,99 @@ def _stats(self):
     def _entropy(self):
         return np.log(2)+1
 
+    def fit(self, data, *args, **kwds):
+        """
+        Maximum likelihood estimate for the location and scale parameters.
+
+        Exact forumulas are used to calculate the loc and scale, unless they
+        are provided in floc or fscale.
+
+        Parameters
+        ----------
+        data : array_like
+            Data to use in calculating the maximum likelihood estimate.
+        floc : float, optional
+            Hold the location parameter fixed to the specified value.
+        fscale : float, optional
+            Hold the scale parameter fixed to the specified value.
+
+        Returns
+        -------
+        loc, scale : float
+            Maximum likelihood estimates for the location and scale.
+
+        Notes
+        -----
+        An error is raised if floc and fscale are given. A warning is raised
+        if either `floc` or `fscale` is provided but does not match the exact
+        MLE value; the generic implementation in rv_continuous, numerical
+        optimization, will be used to determine the best fit.
+        """
+        floc = kwds.pop('floc', None)
+        fscale = kwds.pop('fscale', None)
+
+        check_fit_input_parameters(data, args,
+                                   kwds, fixed_param=(floc, fscale))
+
+        # MLE for the laplace distribution
+        # loc is given by
+        loc = np.median(data)
+
+        # scale is given by
+        scale = (np.sum(np.abs(data - loc))) / len(data)
+
+        # Source: Statistical Distributions, 3rd Edition. Evans, Hastings,
+        # and Peacock (2000), Page 124
+
+        if ((floc is None or np.isclose(floc, loc)) and
+                (fscale is None or np.isclose(fscale, scale))):
+            # Maximum Likelihood Equations used in 3 cases:
+            #   * floc and fscale are not set
+            #   * only floc set, matches MLE equation
+            #   * only fscale set, matches MLE equation
+            return loc, scale
+        else:
+            # Raise warning to indicate optimizer usage for loc or scale
+            if floc is not None:
+                # use optimizer to determine scale for custom floc
+                msg = ("Supplied `floc` did not match the analytical maximum "
+                       "likelihood estimate equation; using numerical "
+                       "optimization to determine `scale`.")
+                warnings.warn(msg, RuntimeWarning)
+                return super(laplace_gen, self).fit(data, floc=floc, *args,
+                                                    **kwds)
+
+            if fscale is not None:
+                # use optimizer to determine loc for custom fscale
+                msg = ("Supplied `fscale` did not match the analytical "
+                       "maximum likelihood estimate equation;"
```

Comment:
```suggestion
                       "maximum likelihood estimate;"
```

Reply:
^These edits are in the commit with the stacklevel change that I haven't pushed until I know that it's the right stacklevel use